pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,138 @@
+package node
+
+import (
+	"reflect"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// LatencyProfileArgVal is used to set arg value pairs for each worker latency profile
+type LatencyProfileArgVal struct {
+	// path to required argument in observed config
+	// eg. observedConfig.extendedArguments.node-monitor-grace-period
+	// is []string{"extendedArguments", "node-monitor-grace-period"}
+	ArgConfigNestedPath []string
+	// value that is to be set for the given argument
+	ArgValue string
+}
+
+// CreateLatencyProfileObserverFn is used to create ObserveConfigFunc that can be used with an configobservation controller to trigger
+// changes to different arg val pairs in observedConfig.* fields and update them on the basis of current worker latency profile
+func CreateLatencyProfileObserverFn(latencyProfileValMap map[configv1.WorkerLatencyProfileType][]LatencyProfileArgVal, conditionFn func() (bool, error)) configobserver.ObserveConfigFunc {
```

Comment:
Actually, we had specific reason in mind behind adding the `conditionFn` to config observer function.

We would want to gate the observer to not update the latency profile arg val pairs in case all the control plane nodes have not achieved the state desired by the current latency profile first. For eg. in case a user is changing the spec from Default to Medium latency profile and then again to the Low profile in a short duration, we would have to reject that update for sometime until the control plane nodes stabilize and we are on the same setting otherwise we could end up breaking the entire cluster. Please refer more about this in [cluster stability analysis which is explained in detail in our worker latency profile EP](https://github.com/openshift/enhancements/blob/master/enhancements/worker-latency-profile/worker-latency-profile.md#cluster-stability-analysis). The same is applicable for both Kube API Server as well as Kube Controller Manager arguments (also for Kubelet), because a mismatch of profiles would essentially result in some nodes having args applied from one latency profile and other nodes having a different profile args which would hamper cluster stability to a great extent.

The plan here was to pass a `conditionFn` in the respective operator that would return if the current operator is in "WorkerLatencyProfileProgressing" state or not. If it's progressing, the `conditionFn` should be `false` and guard against any profile updates, whereas if it's not progressing (i.e. when completed rollout) the func should return `true` allowing the config observer to post new values.

Reply:
Contd. discussion: https://github.com/openshift/library-go/pull/1364#discussion_r866696610, about gating the config observe func whether to rollout new observation values / not.